### DISCUSSION TITLE
Search in relations

### DISCUSSION THREAD

**Discussion by phillipplum** [prisma/prisma#7196] [Q&A] [answered]
Hello,
I need a little help. With the title, I wasn't sure how exactly to bet it. I'll try to describe what I'm going to do.
First of all the models I have:
```prisma
model User {
  id        String @id @default(uuid())
  email     String @unique @db.VarChar(255)

  accessKeys AccessKey[]

  createdAt DateTime     @default(now())
  updatedAt DateTime     @updatedAt
  deletedAt DateTime?
}

model AccessKey {
  id   Int    @id @default(autoincrement())
  name String

  pages     Page[]
  users      User[]

  createdAt DateTime  @default(now())
  updatedAt DateTime  @updatedAt
  deletedAt DateTime?
}

model Page {
  id String @id @default(uuid())

  title            String

  accessKeys AccessKey[]

  createdAt DateTime  @default(now())
  updatedAt DateTime  @updatedAt
  deletedAt DateTime?
}
```

As you can see there are access keys, users and pages. Users "**own**" keys and pages can be "**protected**" by keys.
I would now like to request the following data:

1. All pages that have no keys. Example: This is the case when a "guest" accesses the content or does not have a "key".
2. All pages + pages with the key "Employee". Example: A user has the key "Employee" and can see all pages that do not have a key and that have the key "Employee".

I hope you understand what I mean.

Thank you for your work.

**Comment by ryands17** [comment, marked as answer]
@phillipplum 👋 

> All pages that have no keys. Example: This is the case when a "guest" accesses the content or does not have a "key".

You can do this as follows:

```ts
prisma.page.findMany({ where: { accessKeys: { none: {} } } })
```

> All pages + pages with the key "Employee". Example: A user has the key "Employee" and can see all pages that do not have a key and that have the key "Employee".

This should work:

```ts
prisma.page.findMany({
    where: {
      OR: [
        { accessKeys: { none: {} } },
        { accessKeys: { some: { name: 'Employee' } } },
      ],
    },
})
```

You can read more about these operators here: https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries/#relation-filters

**Reply by phillipplum** [reply]
Okay, I understand! Thank you!
